shell/plugins/global-formatters.js

```diff
@@ -1,10 +1,24 @@
-import Vue from 'vue';
+/* eslint-disable no-console */
+import Vue from 'vue';const components = require.context('@shell/components/formatter', false, /[A-Z]\w+\.(vue)$/);
 
-const components = require.context('@shell/components/formatter', false, /[A-Z]\w+\.(vue)$/);
+const globalFormatters = {
+  install: (Vue) => {
+    components.keys().forEach((fileName) => {
+      const componentConfig = components(fileName);
+      const componentName = fileName.split('/').pop().split('.')[0];
 
-components.keys().forEach((fileName) => {
-  const componentConfig = components(fileName);
-  const componentName = fileName.split('/').pop().split('.')[0];
+      if (Vue.component(componentName)) {
+        // eslint-disable-next-line no-console
+        console.debug(`Skipping ${ componentName } install. Component already exists.`);
+      } else {
+        Vue.component(componentName, componentConfig.default || componentConfig);
+      }
+    });
+  }
+};
 
-  Vue.component(componentName, componentConfig.default || componentConfig);
-});
+export default globalFormatters;
+
+console.warn('The implicit addition of global formatters has been deprecated in Rancher Shell and will be removed in a future version. Make sure to invoke `Vue.use(globalFormatters)` to maintain compatibility.');
+
+Vue.use(globalFormatters);
```

Comment:
It looks like we might only require these lines when running tests. Can we update this to use the same pattern as what we use in the i18n plugin now?

https://github.com/rancher/dashboard/pull/10923/files#diff-4964064b3ea3d66459e9eb386e2612aa8aea93c2d858eca54128348b498dcabb